internal/compiler-interface/src/main/java/xsbti/compile/CachedCompiler2.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright Lightbend, Inc. and Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package xsbti.compile;
+
+import xsbti.AnalysisCallback;
+import xsbti.Logger;
+import xsbti.Reporter;
+import xsbti.VirtualFile;
+import java.io.File;
+
+/**
+ * Define the interface of a cached Scala compiler that can be run.
+ *
+ * This cached compiler hides the implementation of a compiler by just
+ * defining two operations: {@link #commandArguments(File[])} and
+ *
+ */
+public interface CachedCompiler2 extends CachedCompiler {
+  default void run(File[] sources, DependencyChanges changes, AnalysisCallback callback, Logger logger, Reporter delegate, CompileProgress progress) {
+    throw new RuntimeException("run method with Array[File] is not supported for this compiler bridge (see sbt/zinc#829)"); 
```

Comment:
Going to try and send a change to avoid this and the `.asInstanceOf[CachedCompiler2]`.